app/core/src/main/java/com/teamg/tourdeshot/core/mapper/OpeningHoursMapperImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.teamg.tourdeshot.core.mapper;
+
+import com.teamg.tourdeshot.core.api.local.domain.OpeningHoursDTO;
+import com.teamg.tourdeshot.core.mapper.utils.MappingProvider;
+import com.teamg.tourdeshot.core.mapper.utils.OpeningHoursMapperUtils;
+import com.teamg.tourdeshot.core.model.DaySchedule;
+import com.teamg.tourdeshot.core.model.OpeningHours;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.Objects;
+
+@Component
+public class OpeningHoursMapperImpl implements OpeningHoursMapper {
+
+    private final DayScheduleMapper dayScheduleMapper;
+    private final OpeningHoursMapperUtils mapperUtils;
+    private final MappingProvider mappingProvider;
+
+    @Autowired
+    public OpeningHoursMapperImpl(DayScheduleMapper dayScheduleMapper, OpeningHoursMapperUtils mapperUtils, @Qualifier("openStatusMappingsProvider") MappingProvider mappingProvider) {
+        this.dayScheduleMapper = dayScheduleMapper;
+        this.mapperUtils = mapperUtils;
+        this.mappingProvider = mappingProvider;
+    }
+
+    @Override
+    public OpeningHoursDTO toOpeningHoursDTO(OpeningHours openingHours) {
+        if(Objects.isNull(openingHours))
+            return null;
+        LocalDateTime now = LocalDateTime.now();
```

Comment:
When you map list of Locals, each one has different `now`. This can cause different status for each Local, even then have same closing time. You should provide here `now` from higher level, one `now` for each local :)